index.html

```diff
@@ -2175,6 +2232,116 @@ <h3>Verify Proof Sets and Chains</h3>
         </ol>
       </section>
 
+      <section class="normative">
+        <h3>Context Validation</h3>
+
+        <p>
+This section contains an algorithm that applications MUST run after running the
+algorithm in Section [[[#verify-proof]]] or Section
+[[[#verify-proof-sets-and-chains]]], when validating a [=conforming secured
+document=]. This algorithm takes a document ([=map=] |inputDocument|) and a set
+of approved JSON-LD Contexts ([=map=] |approvedContexts|) and returns a [=map=]
+that contains the following:
+        </p>
+
+        <ul>
+          <li>
+a status ([=boolean=] |status|)
+          </li>
+          <li>
+a validated document ([=map=] |document|)
+          </li>
+          <li>
+zero or more warnings ([=list=] of [=ProblemDetails=] |warnings|)
+          </li>
+          <li>
+zero or more errors ([=list=] of [=ProblemDetails=] |errors|)
+          </li>
+        </ul>
+
+        <p>
+The context validation algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set |result|.|status| to `false`.
+          </li>
+          <li>
+For each |contextValue| expressed in every `@context` property, ensure that
+either:
+            <ol class="algorithm">
+              <li>
+The |contextValue| exists in |approvedContexts| and matches a cryptographic hash
+associated with the matching approved context, or
+              </li>
+              <li>
+optionally, for the context values that do not exist in |approvedContexts|,
+remove the |contextValue| from the `@context` property values. Set
+|result|.|document| to the result of running the
+<a data-cite="JSON-LD11-API#compaction-algorithm">
+JSON-LD Compaction Algorithm</a> with the modified |inputDocument| as input.
```

Comment:
This will need the same adjustments as VC data model [PR 1535](https://github.com/w3c/vc-data-model/pull/1535).

Reply:
Done in 0f234e8cd98edd2a245669653ab6bf0d20f09e88 ... which drastically changed the algorithm given @TallTed's changes. The previous algorithm didn't work for multiply nested documents w/ multiple contexts. The update attempts to address that use case.